apps/api/pages/api/event-types/[id]/_auth-middleware.ts

```diff
@@ -2,14 +2,19 @@ import type { NextApiRequest } from "next";
 
 import { HttpError } from "@calcom/lib/http-error";
 
+import { schemaQueryIdAsString } from "~/lib/validations/shared/queryIdString";
 import { schemaQueryIdParseInt } from "~/lib/validations/shared/queryIdTransformParseInt";
 
 async function authMiddleware(req: NextApiRequest) {
   const { userId, isAdmin, prisma } = req;
-  const { id } = schemaQueryIdParseInt.parse(req.query);
+  const idParse = schemaQueryIdParseInt.safeParse(req.query);
+  const { id: slug } = schemaQueryIdAsString.parse(req.query);
+
+  const id = idParse.success ? idParse.data.id : undefined;
+
   if (isAdmin) return;
   const eventType = await prisma.eventType.findFirst({
-    where: { id, users: { some: { id: userId } } },
+    where: { OR: [{ id }, { slug }], AND: { users: { some: { id: userId } } } },
```

Comment:
Because it would either be id or slug and we want to be clear if we are fetching by slug or id so using OR doesn't make sense.

Also, there could possibly be a scenario where an event-type has a numeric slug(say 123). In this case querying for event-types/123 could potentially fetch the eventType with that numeric slug. So, we need to be able to distinguish it clearly in the API request as to what the user wants.

Maybe we could go for /event-types?slug=something or a new endpoint maybe /event-types/bySlug/[slug] ?